OSDOCS-11257 OCP Release Notes 4.14.32 (#78693)
openshift/openshift-docs: release_notes/ocp-4-14-release-notes.adoc

Reviewer: Per the [docs guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-titles-and-section-headings), please add `_{context}` to the end of all these new anchor ids

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

[id="ocp-4-14-32"]
=== RHSA-2024:4329 - {product-title} 4.14.32 bug fix and security updates

Issued: 2024-07-11

{product-title} release 4.14.32, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:4329[RHSA-2024:4329] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2024:4332[RHBA-2024:4332] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.14.32 --pullspecs
----

[id="ocp-4-14-32-enhancements"]
==== Enhancements

The following enhancements are included in this z-stream release:

[id="ocp-4-14-32-pipeline-trigger"]
===== New custom resource definitions for pipeline plugin

> This release updates the pipelines plugin to support the latest Pipeline Trigger API version for the custom resource definitions (CRDs) `ClusterTriggerBinding`, `TriggerTemplate` and `EventListener`. (link:https://issues.redhat.com/browse/OCPBUGS-35723[*OCPBUGS-35723*])

Reviewer: Would just add an [Oxford comma](https://www.ibm.com/docs/en/ibm-style?topic=punctuation-commas#commas-between-items-in-a-series) here:
```suggestion
This release updates the pipelines plugin to support the latest Pipeline Trigger API version for the custom resource definitions (CRDs) `ClusterTriggerBinding`, `TriggerTemplate`, and `EventListener`. (link:https://issues.redhat.com/browse/OCPBUGS-35723[*OCPBUGS-35723*])
```

[id="ocp-4-14-32-_{context}"]
===== Controller for defragmentating etcd

This release introduces a controller to defragment etcd for hosted clusters on Hypershift. (link:https://issues.redhat.com/browse/OCPBUGS-35290[*OCPBUGS-35723*])

[id="ocp-4-14-32-insights-operator-resources_{context}"]
===== Insights Operator collects new resources

> The Insights Operator (IO) now collects the `Prometheus` and `AlertManager` resources in addition to the `openshift-monitoring` custom resource. (link:https://issues.redhat.com/browse/OCPBUGS-36380[*OCPBUGS-36380*])

Reviewer 2: Not sure if "IO" is approved (most things like this aren't). Since it's used one, might as well remove it imo
```suggestion
The Insights Operator now collects the `Prometheus` and `AlertManager` resources in addition to the `openshift-monitoring` custom resource. (link:https://issues.redhat.com/browse/OCPBUGS-36380[*OCPBUGS-36380*])
```

[id="ocp-4-14-32-bug-fixes"]
==== Bug fixes

> * Previously, the `alertmanager-trusted-ca-bundle` configmap was not injected into the user-defined Alertmanager container, which prevented the verification of HTTPS web servers receiving alert notifications. With this update, the trusted CA bundle configmap is mounted into the Alertmanager container at the `/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem` path. (link:https://issues.redhat.com/browse/OCPBUGS-36416[*OCPBUGS-36416*])

Reviewer: I would either split the term out or put in backticks, as shown in the [term glossary](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#config-map):
```suggestion
* Previously, the `alertmanager-trusted-ca-bundle` config map was not injected into the user-defined Alertmanager container, which prevented the verification of HTTPS web servers receiving alert notifications. With this update, the trusted CA bundle configmap is mounted into the Alertmanager container at the `/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem` path. (link:https://issues.redhat.com/browse/OCPBUGS-36416[*OCPBUGS-36416*])
```

> * Previously, for clusters upgraded from older versions of OCP, enabling `kdump` on an OVN-enabled cluster sometimes prevented the node from rejoining the cluster or returning to the `Ready` state. This fix removes problematic stale data from older OCP versions and ensures this kind of stale data is always cleaned up. The node can now start correctly and rejoin the cluster. (link:https://issues.redhat.com/browse/OCPBUGS-36356[*OCPBUGS-36356*])

Reviewer: Change to update per [term glossary](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#update), and spell out OCP on the first mention:
```suggestion
* Previously, for clusters updated from older versions of {product-title} (OCP), enabling `kdump` on an OVN-enabled cluster sometimes prevented the node from rejoining the cluster or returning to the `Ready` state. This fix removes problematic stale data from older OCP versions and ensures this kind of stale data is always cleaned up. The node can now start correctly and rejoin the cluster. (link:https://issues.redhat.com/browse/OCPBUGS-36356[*OCPBUGS-36356*])
```

> * Previously, for clusters updated from older versions of {product-title} (OCP), enabling `kdump` on an OVN-enabled cluster sometimes prevented the node from rejoining the cluster or returning to the `Ready` state. This fix removes problematic stale data from older OCP versions and ensures this kind of stale data is always cleaned up. The node can now start correctly and rejoin the cluster. (link:https://issues.redhat.com/browse/OCPBUGS-36356[*OCPBUGS-36356*])

Reviewer 2: oh - just saw this as I was going to merge. "OPC" is not an approved abbreviation, should just be:
```suggestion
* Previously, for clusters updated from older versions of {product-title}, enabling `kdump` on an OVN-enabled cluster sometimes prevented the node from rejoining the cluster or returning to the `Ready` state. This fix removes problematic stale data from older OCP versions and ensures this kind of stale data is always cleaned up. The node can now start correctly and rejoin the cluster. (link:https://issues.redhat.com/browse/OCPBUGS-36356[*OCPBUGS-36356*])
```

> * Previously, AWS HyperShift clusters leveraged their VPC's primary CIDR range to generate security group rules on the data plane. As a consequence, installing AWS HyperShift clusters into an AWS VPC with multiple CIDR ranges could cause the generated security group rules to be insufficient. With this update, security group rules are generated based on the provided Machine CIDR range instead to resolve this issue. (link:https://issues.redhat.com/browse/OCPBUGS-36159[*OCPBUGS-36159*])

Reviewer: Would spell out the first instance of VPC:
```suggestion
* Previously, AWS HyperShift clusters leveraged their Virtual Private Cloud's (VPC's) primary CIDR range to generate security group rules on the data plane. As a consequence, installing AWS HyperShift clusters into an AWS VPC with multiple CIDR ranges could cause the generated security group rules to be insufficient. With this update, security group rules are generated based on the provided Machine CIDR range instead to resolve this issue. (link:https://issues.redhat.com/browse/OCPBUGS-36159[*OCPBUGS-36159*])
```

* Previously, a bug in `growpart` caused the device to lock which, prevented the LUKS-encrypted devices from opening. The system would fail to boot successfully. With this release, `growpart` was removed from the process and the the system will boot successfully.(link:https://issues.redhat.com/browse/OCPBUGS-35989[*OCPBUGS-35989*])

> * Previously, in User Provisioned Infrastructure (UPI) or clusters that were upgraded from older versions, `failureDomains` may be missing in Infrastructure objects which caused certain checks to fail. With this release, a fallback `failureDomains` is synthesized from `cloudConfig` if none are available in `infrastructures.config.openshift.io`. (link:https://issues.redhat.com/browse/OCPBUGS-35913[*OCPBUGS-35913*])

Reviewer: Same comment about update:
```suggestion
* Previously, in User Provisioned Infrastructure (UPI) or clusters that were updated from older versions, `failureDomains` may be missing in Infrastructure objects which caused certain checks to fail. With this release, a fallback `failureDomains` is synthesized from `cloudConfig` if none are available in `infrastructures.config.openshift.io`. (link:https://issues.redhat.com/browse/OCPBUGS-35913[*OCPBUGS-35913*])
```

> * Previously, in User Provisioned Infrastructure (UPI) or clusters that were upgraded from older versions, `failureDomains` may be missing in Infrastructure objects which caused certain checks to fail. With this release, a fallback `failureDomains` is synthesized from `cloudConfig` if none are available in `infrastructures.config.openshift.io`. (link:https://issues.redhat.com/browse/OCPBUGS-35913[*OCPBUGS-35913*])

Reviewer: Per [ISG word usage](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#m), "may" is reserved for indicating permission, and "might" should be used to indicate possibility instead:
```suggestion
* Previously, in User Provisioned Infrastructure (UPI) or clusters that were upgraded from older versions, `failureDomains` might be missing in Infrastructure objects which caused certain checks to fail. With this release, a fallback `failureDomains` is synthesized from `cloudConfig` if none are available in `infrastructures.config.openshift.io`. (link:https://issues.redhat.com/browse/OCPBUGS-35913[*OCPBUGS-35913*])
```

> * Previously, in User Provisioned Infrastructure (UPI) or clusters that were updated from older versions, `failureDomains` might be missing in Infrastructure objects which caused certain checks to fail. With this release, a fallback `failureDomains` is synthesized from `cloudConfig` if none are available in `infrastructures.config.openshift.io`. (link:https://issues.redhat.com/browse/OCPBUGS-35913[*OCPBUGS-35913*])

Reviewer 2: ```suggestion
* Previously, in user-provisioned infrastructure clusters or clusters that were updated from older versions, `failureDomains` might be missing in Infrastructure objects which caused certain checks to fail. With this release, a fallback `failureDomains` is synthesized from `cloudConfig` if none are available in `infrastructures.config.openshift.io`. (link:https://issues.redhat.com/browse/OCPBUGS-35913[*OCPBUGS-35913*])
```

* Previously, the secrets-store CSI driver on Hypershift was failing to mount secrets due to an issue with the Hypershift CLI. With this release, the driver is able to mount volumes and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-35183[*OCPBUGS-35183*])

* Previously, the reduction of the network queue did not work as expected for inverted rules such as `!ens0`. This happened because the exclamation mark symbol was duplicated in the generated tuned profile. With this release, the duplication no longer occurs so that inverted rules apply as intended. (link:https://issues.redhat.com/browse/OCPBUGS-35012[*OCPBUGS-35012*])

* Previously, a race condition meant the Kubelet might report a false error relating to a failed resize from the volume resizer. With this release, the race condition has been fixed and the false error is no longer printed. (link:https://issues.redhat.com/browse/OCPBUGS-33964[*OCPBUGS-33964*])

> * Previously, escaped strings were not handled properly when editing vSphere connection, causing broken vSphere configuration. With this update, the escape strings work as expected and the vSphere configuration no longer breaks. (link:https://issues.redhat.com/browse/OCPBUGS-33942[*OCPBUGS-33942*])

Reviewer: Seems like there's a missing article here:
```suggestion
* Previously, escaped strings were not handled properly when editing a vSphere connection, causing broken vSphere configuration. With this update, the escape strings work as expected and the vSphere configuration no longer breaks. (link:https://issues.redhat.com/browse/OCPBUGS-33942[*OCPBUGS-33942*])
```

> * Previously, when the default gateway of a node was set to `vlan` and multiple network manager connection had the same name, the node would fail as it could not configure the default OVN-Kubernetes bridge. With this release, the `configure-ovs.sh` shell script includes an `nmcli connection show uuid` command that retrieves the correct network manager connection if many connections with the same name exist. (link:https://issues.redhat.com/browse/OCPBUGS-33590[*OCPBUGS-33590*])

Reviewer: I think connection should be plural here:
```suggestion
* Previously, when the default gateway of a node was set to `vlan` and multiple network manager connections had the same name, the node would fail as it could not configure the default OVN-Kubernetes bridge. With this release, the `configure-ovs.sh` shell script includes an `nmcli connection show uuid` command that retrieves the correct network manager connection if many connections with the same name exist. (link:https://issues.redhat.com/browse/OCPBUGS-33590[*OCPBUGS-33590*])
```
